Downsample/Downsample.cpp

```diff
@@ -106,11 +106,13 @@ Zivid::PointCloud downsample(const Zivid::PointCloud &pointCloud, int downsampli
 
     Zivid::PointCloud pointCloudDownsampled(redDownsampled.rows(), redDownsampled.cols());
 
-    for(int i = 0; i < redDownsampled.rows(); i++)
+    for(size_t i = 0; i < static_cast<size_t>(redDownsampled.rows()); i++)
```

Comment:
Remove the `static_cast`. I do not think there should be any reason to do static_casts or any other casts for that case in this code base at all.

A `static_cast` like this does not change the behavior of your program, the only thing that happens is that you suppress a warning. If you do not care about the warning you should disable it, if you care you need to fix the underlying problem.

Typically when one crank up the warning level in a project developers start to code things that are worse because what they used to do is no longer allowed. And, since they don't get the underlying idea of the warning and they (unintentionally)  hack around it.

I recommend that you refuse to do a change just because a warning tells you to, you need to personally agree that adding the `static_cast<size_t>(` is an improvement. I would argue the only thing this change does is to reduce readability.

I think all in CS team should only do changes they approve of. When you and the warnings are misaligned you need to ask someone what is the reason for having this warning, and what is the correct fix.

Reply:
The good thing with requiring `static_cast` is that you're forced to make a decision whether your narrowing is safe or not. But I agree they make the code uglier, and these warnings might be too strict for this repo. I'm fine with disabling narrowing warnings. 

Reply:
It is int again.